src/Microsoft.AspNetCore.Razor.Runtime/Resources.resx

```diff
@@ -195,4 +195,16 @@
   <data name="ArgumentMustBeAnInstanceOf" xml:space="preserve">
     <value>Argument must be an instance of '{0}'.</value>
   </data>
+  <data name="TagHelperDescriptorFactory_CouldNotFindMatchingEndBrace" xml:space="preserve">
+    <value>Could not find matching ']' for required attribute.</value>
```

Comment:
This message sounds like it's describing an internal inadequacy of our code. Tell the user what's wrong with their `HtmlTargetElementAttribute.Attributes` value (and pass in the property and class names).